security/pkg/pki/ca/ca_test.go

```diff
@@ -373,140 +373,193 @@ func TestCreatePluggedCertCA(t *testing.T) {
 // TODO: merge tests for SignCSR.
 func TestSignCSRForWorkload(t *testing.T) {
 	subjectID := "spiffe://example.com/ns/foo/sa/bar"
-	opts := util.CertOptions{
-		// This value is not used, instead, subjectID should be used in certificate.
-		Host:       "spiffe://different.com/test",
-		RSAKeySize: 2048,
-		IsCA:       false,
-	}
-	csrPEM, keyPEM, err := util.GenCSR(opts)
-	if err != nil {
-		t.Error(err)
-	}
-
-	ca, err := createCA(time.Hour)
-	if err != nil {
-		t.Error(err)
-	}
-
-	requestedTTL := 30 * time.Minute
-	certPEM, signErr := ca.Sign(csrPEM, []string{subjectID}, requestedTTL, false)
-	if signErr != nil {
-		t.Error(err)
-	}
-
-	fields := &util.VerifyFields{
-		ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
-		KeyUsage:    x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment,
-		IsCA:        false,
-		Host:        subjectID,
-	}
-	_, _, certChainBytes, rootCertBytes := ca.GetCAKeyCertBundle().GetAll()
-	if err = util.VerifyCertificate(
-		keyPEM, append(certPEM, certChainBytes...), rootCertBytes, fields); err != nil {
-		t.Error(err)
-	}
-
-	cert, err := util.ParsePemEncodedCertificate(certPEM)
-	if err != nil {
-		t.Error(err)
+	cases := map[string]struct {
```

Comment:
The majority of the test files for this PR that were altered had little actual changes. I made them largely table tests to add support for EC unit tests. As a result, however, the diffs are larger due to indentation.